cli/lock.ml

```diff
@@ -84,25 +84,41 @@ let is_duniverse_repo repo_url =
   let url = OpamUrl.to_string repo_url in
   String.equal url D.Config.duniverse_opam_repo
 
-let check_dune_universe_repo ~repositories non_dune_packages =
+let error_message_when_dependencies_don't_build_with_dune ~repositories
+    non_dune_packages =
   let dune_universe_is_configured =
     List.exists ~f:is_duniverse_repo repositories
   in
-  if not dune_universe_is_configured then
-    Logs.warn (fun l ->
-        l
-          "Couldn't calculate a set of packages to satisfy the request. Note \
-           that %a will fail if not all of the project dependencies use dune \
-           as their build system, in your project that would be %a. To solve \
-           this issue there exists a dune-universe opam-repository which \
-           contains dune ports for some opam packages, but it is currently not \
-           set in your current switch. If you wish to set it up, run the \
-           following command:\n\
-           opam repository add dune-universe %s"
-          Fmt.(styled `Bold string)
-          "opam monorepo lock"
-          Fmt.(list ~sep:comma D.Opam.Pp.package_name)
-          non_dune_packages D.Config.duniverse_opam_repo)
+  let dune_universe_state_message =
+    if dune_universe_is_configured then
+      Fmt.str
+        "The dune-universe opam repository (%s) contains dune ports of some \
+         popular packages to help build more packages with dune however it \
+         appears to already be set up on this switch. Thus it is possible that \
+         no dune port exists for any of these packages."
+        D.Config.duniverse_opam_repo
+    else
+      Fmt.str
+        "The dune-universe opam repository (%s) contains dune ports of some \
+         popular packages to help build more packages with dune. It doesn't \
+         appear to be set up on this switch. Adding it to this switch may fix \
+         this issue. Add the dune-universe opam repository to this switch by \
+         running the command:\n\n\
+         opam repository add dune-universe %s" D.Config.duniverse_opam_repo
+        D.Config.duniverse_opam_repo
+  in
+  Fmt.str
+    "Some dependencies cannot be built with dune!\n\n\
+     opam-monorepo requires that all dependencies use dune as their build \
+     system.\n\n\
+     These dependencies (possibly transitive) don't use dune as their build \
+     system:\n\
+     %a\n\n\
+     %s"
+    Fmt.(
+      list ~sep:(const char '\n') (fun ppf p ->
+          Fmt.pf ppf "- %a" D.Opam.Pp.package_name p))
```

Comment:
For (marginally) better readability, maybe make this a separate function, `pp_package_entry` or something like this?